### PR TITLE
week-3 assignments

### PR DESCRIPTION
**Pull request by addisdave**
# Submitting Week 3 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Interview Questions
- [x] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: react-debugging.html [hunk 1]
```diff
@@ -81,6 +87,8 @@ <h1>Draw a Card</h1>
               )
             })}
           </div>
+         </>
+
         )
       }
```
Comment (SunkissedQueen) [inline react-debugging.html:94]: nice

File: interview-questions.md [hunk 2]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Researched answer: state holds information about the component therefor when a component need to track of information between rendering,it uses state. props is a React keyword used for data transfer between components. 
 
 6. STRETCH: Which is the difference between a div and a span?
 
-Your answer:
+Your answer: div is used for CSS based HTML layouts while span is used to sylise text. 
 
 Researched answer:
 
 ## Looking Ahead: Terms for Next Week
```
Comment (SunkissedQueen) [inline interview-questions.md:55]: looking forward to these lecture

File: interview-questions.md [hunk 3]
```diff
@@ -8,48 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
```
Comment (SunkissedQueen) [inline interview-questions.md:8]: Consider what you want to incorporate into your interviews. Create a flow for yourself and share it on next week assessment. Things I am looking for more content to include an example of how you used the topic. Think making a conversation with the interviewer.

File: code-challenges.test.js [hunk 4]
```diff
@@ -43,13 +128,25 @@ const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
...
 
-const numbersToAdd2 = [0, 7, -8, 12]
+/ RED
+// 1- Write the test
+// 2- Run a test "yarn jest" to see failed test.
+
+// GREEN
+// 3- Write the code 
+
+//Psudocode:
```
Comment (SunkissedQueen) [inline code-challenges.test.js]: please include what information you need to get from the input to the output

File: code-challenges.test.js [hunk 5]
```diff
@@ -11,28 +11,113 @@
...
+// Write a function called fibLength1
+// Parameter is Fibonacci numbers of 6.
+// Expected out put of an array containg the following numbers [1, 1, 2, 3, 5, 8]
+
+//const fibLength1 = 6
 // Expected output: [1, 1, 2, 3, 5, 8]
```
Comment (SunkissedQueen) [inline code-challenges.test.js]: Look over the structure of functions. Use a parameter vice the argument. The parameter will allow the function to be reusable.

File: code-challenges.test.js [hunk 6]
```diff
@@ -11,28 +11,113 @@
...
+     myFibs.push(c)
+     
+} 
+console.log(fibLength1.myFibs())
+
+
+
+// *** couldl not go further than this on the coding challeges *** On the yarn jest run test, I kept getting
+// the jason.file not found error eventhogh i am in the right directory (repositoriy)
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js]: Are the node modules, package.json, and other files in the same directory?

**Review by SunkissedQueen — CHANGES_REQUESTED** [review]
Dawit, I see that there are some blockers on the code challenges. We can attempt to tackle those this morning. I will need to see all the tests and functions to complete this review. Suggestions for the interview questions: remember it is a conversation. Ensure you are not only giving definitions but also examples to show how you have used the tool.

**Comment by addisdave** [comment]
check

File: code-challenges.test.js [hunk 7]
```diff
@@ -10,47 +10,150 @@
...
+// number
+//   ✕ Create a function that takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence. (1 ms)
+// oddNum
+//   ✕ takes in an array and returns a new array of only odd numbers sorted from least to greatest
+// sumArray
+//   ✕ takes in an array and returns an array of the accumulating sum. An empty array should return an empty array. (1 ms)
 
 // b) Create the function that makes the test pass.
+
+const sumArray = (array) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:154]: Let's think about what you want to do with the array. What information do you need to get from the input to the expected output?

**Review by SunkissedQueen — APPROVED** [review]
Dawit, you are approved. Thanks for working through these test and functions. As you see with practice, it becomes smoother. Keep a template of the test and then refactor it according to the question. The interview answers need more structure and content. The interview questions is a time for you to practice what you would do at an interview. Set up a flow: definition and examples.

**Pull request merged**